finder.ts

```diff
@@ -20,6 +20,8 @@ export type Options = {
   optimizedMinLength: number
   threshold: number
   maxNumberOfTries: number
+  timeoutMs: number
```

Comment:
I see that the input to finder is using Partial<Options> but I think we can possibly use undefined for timeoutMs as well.

Reply:
Let me try that out! -1 is less straight-forward